packages/pg-v5/lib/psql.js

```diff
@@ -1,107 +1,247 @@
 'use strict'
 
-const co = require('co')
+const { once, EventEmitter } = require('events')
+
 const bastion = require('./bastion')
 const debug = require('./debug')
 
-function handlePsqlError (reject, psql) {
-  psql.on('error', (err) => {
-    if (err.code === 'ENOENT') {
-      reject(`The local psql command could not be located. For help installing psql, see https://devcenter.heroku.com/articles/heroku-postgresql#local-setup`)
+function psqlQueryOptions (query, dbEnv) {
```

Comment:
Rather than opening a process with `child_process.spawn` handling the `data` and `error` events in every function, I turned `psqlExec`, `psqlExecFile`, and `psqlExecInteractive` into functions that return options for spawn instead. This lets us keep all the event handler logic in one place and reduces duplication throughout the 3 ways we allow customers to run `psql` through `pg:psql`